lib/matplotlib/streamplot.py

```diff
@@ -464,30 +461,27 @@ def integrate(x0, y0):
         resulting trajectory is None if it is shorter than `minlength`.
         """
 
-        stotal, x_traj, y_traj = 0., [], []
+        stotal, xy_traj = 0., []
 
         try:
             dmap.start_trajectory(x0, y0)
         except InvalidIndexError:
             return None
         if integration_direction in ['both', 'backward']:
-            s, xt, yt = _integrate_rk12(x0, y0, dmap, backward_time, maxlength)
+            s, xyt = _integrate_rk12(x0, y0, dmap, backward_time, maxlength)
             stotal += s
-            x_traj += xt[::-1]
-            y_traj += yt[::-1]
+            xy_traj += xyt[::-1]
 
         if integration_direction in ['both', 'forward']:
             dmap.reset_start_point(x0, y0)
-            s, xt, yt = _integrate_rk12(x0, y0, dmap, forward_time, maxlength)
-            if len(x_traj) > 0:
-                xt = xt[1:]
-                yt = yt[1:]
+            s, xyt = _integrate_rk12(x0, y0, dmap, forward_time, maxlength)
             stotal += s
-            x_traj += xt
-            y_traj += yt
+            xy_traj += xyt[1:]
 
-        if stotal > minlength:
-            return x_traj, y_traj
+        if stotal > minlength > 0:
+            return [*zip(*xy_traj)]
+        elif minlength == 0:
+            return [], []
```

Comment:
Overall, I'm +/-0 on packing x, y together. It is shorter in some places, but for example this construct is much harder to understand.

Reply:
The way to make this actually simpler is to make `get_integrator` return an (N, 2) array rather than a (2, N) array which is also consistent with what we do everywhere else, but basically means switching to a private `_get_integrator` with the new convention, so let's also deprecate the old one.